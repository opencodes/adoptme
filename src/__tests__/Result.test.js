import { expect, test, describe } from '@jest/globals';
import { create } from 'react-test-renderer'
import { StaticRouter } from "react-router-dom";

import Results from '../Results'
const pets = [{
    id: 1,
    name: "Luna",
    animal: "dog",
    city: "Seattle",
    state: "WA",
    description:
        "Luna is actually the most adorable dog in the world. Her hobbies include yelling at squirrels, aggressively napping on her owners' laps, and asking to be fed two hours before IT'S DAMN WELL TIME LUNA. Luna is beloved by her puppy parents and lazily resides currently in Seattle, Washington.",
    breed: "Havanese",
    images: [
        "http://pets-images.dev-apis.com/pets/dog25.jpg",
        "http://pets-images.dev-apis.com/pets/dog26.jpg",
        "http://pets-images.dev-apis.com/pets/dog27.jpg",
        "http://pets-images.dev-apis.com/pets/dog28.jpg",
        "http://pets-images.dev-apis.com/pets/dog29.jpg",
    ],
},]
describe('Results', () => {
    test('should render with no pets', () => {
        const tree = create(<Results pets={[]} />).toJSON();
        expect(tree).toMatchSnapshot()
    })
    test('should render with no pets', () => {
        const tree = create(<StaticRouter><Results pets={pets} /></StaticRouter>).toJSON();
        expect(tree).toMatchSnapshot()
    })
})
