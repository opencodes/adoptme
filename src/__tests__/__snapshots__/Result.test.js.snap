// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should render with no pets 1`] = `
<div
  className="search"
>
  <h1>
    No Pets Found
  </h1>
</div>
`;

exports[`Results should render with no pets 2`] = `
<div
  className="search"
>
  <a
    className="pet"
    href="/details/1"
    onClick={[Function]}
  >
    <div
      className="image-container"
    >
      <img
        alt="Luna"
        data-testid="thumbnail"
        src="http://pets-images.dev-apis.com/pets/dog25.jpg"
      />
    </div>
    <div
      className="info"
    >
      <h1>
        Luna
      </h1>
      <h2>
        dog — Havanese — Seattle, WA
      </h2>
    </div>
  </a>
</div>
`;
